Move Simple Payments to PayPal Payments package

allie500 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

Fixes PAYPAL-13

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* This PR moves the Simple Payments block code to the PayPal Payments package.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

***Reviewers: Please check Slack. I have sent you a message with the PayPal Sandbox test account credentials for testing this PR.***

* Test this on the following sites:
  * WoA site without the business plan
  * WoA site with the business plan
  * JN site without Jetpack Complete plan
  * JN site with Jetpack Complete plan
  * Simple site without the business plan
  * Simple with the business plan
* Use Jetapack Beta plugin to checkout this branch. 

For each site check the following:

* Is the 'Pay with PayPal' block available in the editor? 
  * It should be for sites with plans.
  * It should not be for sites without plans.
* For sites with plans: 
  * In the editor, can you make changes to the product and save? Do the changes persist?
  * Add the PayPal Sandbox business email address in the block's address field for later payment testing.
  * Does the block render correctly on the front end?
  * If you click the 'Pay with PayPal' button does the PayPal modal successfully load?

It is possible to test payments using the PayPal Sandbox with the Simple Payments block. However, it is currently only possible to test payments on WoA and JN sites. Simple sites encounter a CORS issue when sandboxed, preventing payment testing regardless of the version of Jetpack running on the sandbox.

To test payments, do the following:
* Visit your WoA or JN site.
* Open the browser inspector's application tab
* Under local storage, select your site's URL.
* Then add the following key - value pair: 
  * Key: `simple-payments-env`
  * Value: `sandbox`
* Refresh the page, you should now see `https://www.sandbox.paypal` in the list of sites in local storage in the browser inspector application tab.
* Navigate to the frontend page on your site with a Simple Payment block on it. 
* Click the Pay with PayPal button.
* Add the PayPal Sandbox customer test account email address to the email field in the PayPal modal.
* Click Next
* Then click the Use Password Instead button.
* Enter the password that you were provided in Slack.
* Click login.
* Then click Pay.
* Wait a few moments (5-10 seconds). The modal should disappear.
* You should see a success message show within the Simple Payments block
* You should receive an email at the site's admin email address notifying you of a successful sale.

allie500 committed:
Move simple payment block to pkg

allie500 committed:
changelog

allie500 committed:
Fix block.json textdomain

allie500 commented:
@jeherve, so far I've only moved and lightly modified the Simple Payments block code. I see there is also a `simple-payments` directory within the `jetpack/projects/plugins/jetpack/modules` directory:
https://github.com/Automattic/jetpack/tree/update/move-simple-payments-to-paypal-payments-pkg/projects/plugins/jetpack/modules/simple-payments

Should I make a `modules` directory in `projects/packages/paypal-payments` and move the three files there? Or would it be better to simply move the three files directly into the `projects/packages/paypal-payments` directory?

Thanks in advance for your feedback!

jeherve commented:
That's going to be a multi-step process I think. Here is how I would personally go about it, but that's just my personal opinion, feel free to take a different approach :)

The main entry point for the whole functionality is `projects/plugins/jetpack/modules/simple-payments/simple-payments.php`. I would recommend copying this to the package first, alongside the related files in that `projects/plugins/jetpack/modules/simple-payments/` directory.

Once that's ported to the package, we can start requiring the package in the Jetpack plugin's `composer.json` file, stop requiring the file here:
https://github.com/Automattic/jetpack/blob/5661c6259654a343b52c5cb1e0cfc8dfd3a401f3/projects/plugins/jetpack/modules/module-extras.php#L37
And instead create a new `modules/simple-payments.php` file that would instantiate the class from the package.

Once that's done, you could copy the contents of `projects/plugins/jetpack/modules/widgets/simple-payments.php` to the package, and then replace the contents of that widget file by a call to the same code in the package.

Then, the next step would be to copy the actual block from `projects/plugins/jetpack/extensions/blocks/simple-payments/` into a new dir in the package, and create the webpack config file to build it, so you get build files that we'll be able to require in the Jetpack plugin and later on in the new standalone plugin. The Forms package has an example you could follow. Here is their webpack file for example: `projects/packages/forms/tools/webpack.config.blocks.js`.
Once you have that set up, you would probably create a new class in the package, that would handle registering that block. You can see the `Contact_Form_Block` for an example of a block that lives in a package, is registered in a package, and then registration is called from the Jetpack plugin in `projects/plugins/jetpack/extensions/blocks/contact-form/contact-form.php`.

Once that's all done, you'll be able to deprecate the old classes from the Jetpack plugin, and they could then be removed entirely in a few releases.

At that point, you'll have a standalone package with code that can register a block, register a widget, and you'll be able to register that block or that widget from the Jetpack plugin and also from any other standalone plugin we create in the future.

Let me know if that makes sense.

allie500 committed:
Revert all commits in PR.

This reverts commits:
- cacc7eb39f9c97f7332e23c72c51acc169ffe11f
- 78e3c919830946e71bb7ac28785fff3085eac34a
- b3eed5ca43e3ff948f6474fefb91ac5c7027ecfe

allie500 committed:
changelog

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 commented:
@jeherve I'm finally getting back over to this. Thanks so much for the detailed guidance. I'm going to follow that to get this moving forward.

allie500 committed:
Move simple payments module files to package

allie500 committed:
Update Jetpack plugin composer.json

allie500 committed:
Remove simple-payments from module-extras

allie500 committed:
Add simple-payments module file

allie500 committed:
changelog

allie500 commented:
> That's going to be a multi-step process I think. Here is how I would personally go about it, but that's just my personal opinion, feel free to take a different approach :)
> 
> The main entry point for the whole functionality is `projects/plugins/jetpack/modules/simple-payments/simple-payments.php`. I would recommend copying this to the package first, alongside the related files in that `projects/plugins/jetpack/modules/simple-payments/` directory.
> 
> Once that's ported to the package, we can start requiring the package in the Jetpack plugin's `composer.json` file, stop requiring the file here:
> 
> https://github.com/Automattic/jetpack/blob/5661c6259654a343b52c5cb1e0cfc8dfd3a401f3/projects/plugins/jetpack/modules/module-extras.php#L37
> 
> 
> And instead create a new `modules/simple-payments.php` file that would instantiate the class from the package.

@jeherve I have completed all of the above. At this point, I have tried running the PHP unit tests for Jetpack plugin just to make sure everything is working. The tests fail to run with the error:

```
Verifying lock file contents can be installed on current platform.
Warning: The lock file is not up to date with the latest changes in composer.json. You may be getting outdated dependencies. It is recommended that you run `composer update` or `composer update <package name>`.
- Required package "automattic/jetpack-paypal-payments" is not present in the lock file.
```

I think at this point, I would need to publish the package to Packagist as the lock file doesn't get updated if I run `composer update`. I would publish using the [PayPal Payments mirror repo](https://github.com/Automattic/jetpack-paypal-payments), correct?

allie500 committed:
Properly update Jetpack plugin's composer.json
Update lock file.

allie500 commented:
I just realized I totally skipped a command to properly add the package to the Jetpack plugin's `composer.json` file. I've run that and now I've got the lock file updated as well.

jeherve commented on projects/packages/paypal-payments/class-simple-payments.php:
I would recommend against moving the files ; I would suggest copying instead, and marking the different classes and methods that are left in the Jetpack plugin as deprecated with `_deprecated_function`. It will make it easier to spot anything we've left behind, and it will also ensure we do not create fatal errors on plugin update, for sites that have aggressive caching setups and that may continue to require a file that no longer exists.

allie500 replied:
Ok. Should I revert all my commits from yesterday and re-do all the steps but instead of moving the files, just copy them over and modify both sets as needed?

For the old ones mark them with `_deprecated_function`, and the new files in the package make needed changes to register assets, update constants, etc etc.

allie500 replied:
I added the `simple-payments` module directory and the three files within it in [72901a9](https://github.com/Automattic/jetpack/pull/43413/commits/72901a9491df13d6731ab2b1e11a610f6fe5c260).

I deprecated all the public functions in `projects/plugins/jetpack/modules/simple-payments/simple-payments.php` in [5ece6e7](https://github.com/Automattic/jetpack/pull/43413/commits/5ece6e7cbb4cbae4fd9557ef6d5872f17d8e2484).

jeherve commented on projects/plugins/jetpack/modules/simple-payments.php:
```diff
@@ -0,0 +1,17 @@
+ * @package automattic/jetpack
+ */
+
+require_once __DIR__ . '/../packages/paypal-payments/class-simple-payments.php';
```
That won't work on production since the required package will not live in this directory in the production version of the plugin. 

Since the package will be autoloaded, you do not have to worry about requiring the file.

allie500 replied:
I've removed the `require_once` statement in [0169271](https://github.com/Automattic/jetpack/pull/43413/commits/0169271ff88d91506a7733109c5dc80c488064fd).

That said, do I need to add a `use` statement to import the correct one? After removing the `require_once` my IDE is showing it using the `Jetpack_Simple_Payments::get_instance();` from the deprecated class.

allie500 replied:
I've added that `use` statement to get it to use the correct class in [0b47eca](https://github.com/Automattic/jetpack/pull/43413/commits/0b47eca3388467ff407304dbee4dd219de023844).

jeherve commented on projects/plugins/jetpack/modules/module-extras.php:
```diff
@@ -34,7 +34,6 @@
 	'external-media/external-media.php',
 	'plugin-search.php',
 	'scan/scan.php', // Shows Jetpack Scan alerts in the admin bar if threats found.
-	'simple-payments/simple-payments.php',
```
Here I would recommend that you replace loading the file by loading the module file, `'simple-payments.php'`.

allie500 replied:
Thanks for catching this! Updated in [a042df9](https://github.com/Automattic/jetpack/pull/43413/commits/a042df939bcb53a7ddfbd183a77790c4f85509c3).

jeherve reviewed (commented):
> I think at this point, I would need to publish the package to Packagist as the lock file doesn't get updated if I run composer update. I would publish using the [PayPal Payments mirror repo](https://github.com/Automattic/jetpack-paypal-payments), correct?

I took care of that in the last PR, when you introduced the package skeleton. The package lives here:
https://packagist.org/packages/automattic/jetpack-paypal-payments

> UndefError PhanUndeclaredConstant Reference to undeclared constant \JETPACK__VERSION

Now that this lives in the package, you can use the package version (`PACKAGE_VERSION`) instead.

> plugins_url

You may have to make some changes where scripts and styles are registered, since the files are no longer in the same location. `Assets::register_script()` can be a good replacement at times.

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Re-add simple-payments module dir and files

allie500 committed:
Add namespace and imports

allie500 committed:
Deprecate old Jetpack_Simple_Payments class

allie500 committed:
Remove unneeded require_once

allie500 committed:
Add new module file to module-extras.php

allie500 committed:
Update Simple Payments module loader.

allie500 committed:
Add PACKAGE_VERSION
Update style and script registration

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Rename Jetpack_Simple_Payments to Simple_Payments

allie500 committed:
Create Simple_Payments_Widget in pkg

allie500 committed:
Update class reference

allie500 committed:
Update Jetpack plugin's simple payments widget class

jeherve commented:
As we discussed in Slack:

- Moving the new files into `src` will help.
- Since you're introducing new files, you will indeed need to fix those issues reported by Phan, and then run `jetpack phan --update-baseline packages/paypal-payments`.

allie500 committed:
Move files to src subdirs

jeherve commented on projects/packages/paypal-payments/src/legacy/class-simple-payments.php:
```diff
@@ -0,0 +1,834 @@
+
+namespace Automattic\Jetpack\Paypal_Payments;
+
+use Automattic\Jetpack\Assets;
+use Automattic\Jetpack\Current_Plan as Jetpack_Plan;
```
Since you're using specific packages inside the PayPal Payments package, you would need to add them as dependencies to the PayPal Payments package.

allie500 replied:
Thanks for catching this. Addressed in [ceccdd8](https://github.com/Automattic/jetpack/pull/43413/commits/ceccdd80a7e31010bb647fcb50a390915e2e6b01).

jeherve commented on projects/packages/paypal-payments/src/legacy/class-simple-payments.php:
```diff
@@ -0,0 +1,834 @@
+
+use Automattic\Jetpack\Assets;
+use Automattic\Jetpack\Current_Plan as Jetpack_Plan;
+use Jetpack;
+use Jetpack_Components;
+use Jetpack_Currencies;
```
Those Jetpack-the-plugin specific dependencies aren't great. That means you won't be able to use that class on a site that doesn't run the Jetpack plugin. 

I think we'll need to replace them by functionality available in packages, or move those classes to packages to be able to use them.

allie500 replied:
As discussed in Slack, I've added a check in `get_instance()` to bail if this classes don't exist in [302ea6d](https://github.com/Automattic/jetpack/pull/43413/commits/302ea6d996a5ed98290ed5b87eda96a1e200028e).

jeherve commented on projects/packages/paypal-payments/src/legacy/class-simple-payments.php:
```diff
@@ -0,0 +1,834 @@
+ * Simple_Payments
+ */
+class Simple_Payments {
+	// These have to be under 20 chars because that is CPT limit.
```
This seems to be a left-over from something else.

allie500 replied:
It was. I forgot to remove it. Removed in [83ee3d0](https://github.com/Automattic/jetpack/pull/43413/commits/83ee3d07174550f9d18b4a8654f5644b20fe66f4).

jeherve commented on projects/packages/paypal-payments/src/legacy/class-simple-payments.php:
```diff
@@ -0,0 +1,834 @@
+	 */
+	private function __construct() {}
+
+	/**
+	 * Original singleton.
+	 *
+	 * @todo Remove this when nothing calles getInstance anymore.
+	 *
+	 * @deprecated 10.8
+	 */
+	public static function getInstance() { // phpcs:ignore WordPress.NamingConventions.ValidFunctionName.MethodNameInvalid
+		_deprecated_function( __METHOD__, 'Jetpack 10.7.0', 'Simple_Payments::get_instance' );
+		return self::get_instance();
+	}
```
Do we need this in the new class in the package?

allie500 replied:
We don't. I removed it in [83ee3d0](https://github.com/Automattic/jetpack/pull/43413/commits/83ee3d07174550f9d18b4a8654f5644b20fe66f4).

jeherve commented on projects/packages/paypal-payments/src/legacy/class-simple-payments.php:
```diff
@@ -0,0 +1,834 @@
+class Simple_Payments {
+	// These have to be under 20 chars because that is CPT limit.
+
+	const PACKAGE_VERSION = '0.1.0-alpha';
```
Can we use the `PACKAGE_VERSION` from the main `PayPal_Payments` class here?

allie500 replied:
Yes we should do that. Thanks for catching this. Addressed in [83ee3d0](https://github.com/Automattic/jetpack/pull/43413/commits/83ee3d07174550f9d18b4a8654f5644b20fe66f4).

jeherve commented on projects/packages/paypal-payments/src/legacy/class-simple-payments.php:
```diff
@@ -0,0 +1,834 @@
+		}
+
+		return ( ( defined( 'IS_WPCOM' ) && IS_WPCOM )
+			|| Jetpack::is_connection_ready() )
```
You should be able to switch to relying on the Connection package for this (and thus require the connection package in the PayPal Payments package).

allie500 replied:
Thanks for flagging this and sharing the Connection package. I've added the package as a dependency in [1632a85](https://github.com/Automattic/jetpack/pull/43413/commits/1632a85f6ded84bcff4834646532227eb6a77729).

I've replaced the method call in [e784072](https://github.com/Automattic/jetpack/pull/43413/commits/e78407258634a0d92750324ad015a7eee5999b46).

jeherve commented on projects/packages/paypal-payments/src/legacy/class-simple-payments.php:
```diff
@@ -0,0 +1,834 @@
+		$data['id'] = $attrs['id'];
+
+		if ( ! $this->is_enabled_jetpack_simple_payments() ) {
+			if ( jetpack_is_frontend() ) {
```
this is a Jetpack-the-plugin only function. I think you'll need to move it to a package to start using it here.

allie500 replied:
I've added a check in `get_instance()` for this function. If it doesn't exist it returns null. Added in [302ea6d](https://github.com/Automattic/jetpack/pull/43413/commits/302ea6d996a5ed98290ed5b87eda96a1e200028e).

allie500 committed:
Copy simple payments block code to pkg

allie500 committed:
Add block dependencies

allie500 committed:
Add webpack config

allie500 committed:
Copy register-jetpack-block to pkg

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Add assets and plans packages as deps

allie500 committed:
Remove left over comment
Update PACKAGE_VERSION value
Remove old deprecated method

allie500 committed:
Add Jetpack Connection dependency

allie500 committed:
Replace Jetpack method with Connection method.

allie500 committed:
Update composer.json build scripts

allie500 committed:
Add dependencies check in get_instance

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Update pkg dependencies

allie500 commented:
Copying the Simple Payments block code to the package in this [commit](https://github.com/Automattic/jetpack/pull/43413/commits/b5d0cedce6b4918dc46281f6293470ca7c11dcba) is the cause of unrelated extensions' JS tests to fail. I haven't been able to track down the root cause.

allie500 committed:
Match JS deps with Jetpack plugin

anomiex commented on projects/packages/paypal-payments/package-lock.json:
This should be removed. It's an npm lockfile, and we use pnpm.

allie500 replied:
Thanks for flagging it. I probably accidentally ran `npm` at some point instead of `pnpm`. 😅

allie500 replied:
Removed in [d12889d](https://github.com/Automattic/jetpack/pull/43413/commits/d12889db71a3ea1f04c5a0c6ef914af489c9ddd5).

allie500 commented:
> Copying the Simple Payments block code to the package in this [commit](https://github.com/Automattic/jetpack/pull/43413/commits/b5d0cedce6b4918dc46281f6293470ca7c11dcba) is the cause of unrelated extensions' JS tests to fail. I haven't been able to track down the root cause.

I got some assistance to figure out my error. Some of the dependencies in the package didn't match the dependencies in the the Jetpack plugin. This caused issues. I've addressed it in [ff0eb51](https://github.com/Automattic/jetpack/pull/43413/commits/ff0eb51e8af18b2d8458a920a96a2d72e027af17). Tests are passing for me locally now.

allie500 committed:
Remove unneeded npm lock file

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Fix text domain in block.json

allie500 committed:
Update Jetpack composer.lock file

jeherve commented on projects/packages/paypal-payments/composer.json:
```diff
@@ -4,7 +4,10 @@
+		"php": ">=7.2",
+		"automattic/jetpack-assets": "@dev",
+		"automattic/jetpack-plans": "@dev",
+		"automattic/jetpack-connection": "@dev"
```
You'll need to require `automattic/jetpack-blocks` as well since you're using it in the package.

allie500 replied:
I've added this in [da31fc6](https://github.com/Automattic/jetpack/pull/43413/commits/da31fc6110df737a802f9d3fc51eb1b4c1f524be).

jeherve commented on projects/packages/paypal-payments/composer.json:
```diff
@@ -18,8 +21,12 @@
 	"scripts": {
-		"build-development": "echo 'Add your build step to composer.json, please!'",
-		"build-production": "echo 'Add your build step to composer.json, please!'",
+		"build-production": [
+			"pnpm run build-production"
+		],
+		"build-development": [
+			"pnpm run build"
+		],
```
This will not be useful as long as you haven't implemented a build step in the package's `package.json`, to build the block.

allie500 replied:
I've added an initial build command in the `package.json` file in [515815e](https://github.com/Automattic/jetpack/pull/43413/commits/515815e86ee65b6388567ec3e481062a85966c84).

jeherve commented on projects/packages/paypal-payments/src/register-jetpack-block.js:
```diff
@@ -0,0 +1,89 @@
+import {
```
I think it may be worth adding a comment here to mention that this is a copy of another file. Maybe with a to-do suggesting the move to the `@automattic/jetpack-shared-extension-utils` package.

allie500 replied:
Thanks for this suggestion! I've added it in [ceb7193](https://github.com/Automattic/jetpack/pull/43413/commits/ceb719327f34cec855456317bfa1344d11d103ad).

jeherve commented on projects/packages/paypal-payments/src/register-jetpack-block.js:
Should this be in the `block` directory?

allie500 replied:
I'm not sure. I can move it there if you think it should go there.

jeherve replied:
I think it would make sense for this to be an utility within the block directory.

allie500 replied:
Got it. I've moved it in [124408b](https://github.com/Automattic/jetpack/pull/43413/commits/124408b3c4b7d7b69e71102fe329ed1b5c8b6680).

jeherve commented on projects/packages/paypal-payments/package.json:
```diff
@@ -21,5 +21,19 @@
 	},
-	"devDependencies": {}
+	"dependencies": {
+		"@automattic/jetpack-shared-extension-utils": "workspace:*",
```
Should we add a dependency to , since that's needed for the build?

```suggestion
		"@automattic/jetpack-shared-extension-utils": "workspace:*",
		"@automattic/jetpack-webpack-config": "workspace:*",
```

allie500 replied:
Thanks for flagging this. I've added it in [515815e](https://github.com/Automattic/jetpack/pull/43413/commits/515815e86ee65b6388567ec3e481062a85966c84).

jeherve commented on projects/plugins/jetpack/modules/simple-payments.php:
```diff
@@ -0,0 +1,14 @@
+<?php
+/**
+ * Simple Payments module loader.
```
It may be worth adding a comment saying that this file is not a typical module ; the file is required from `module-extras.php`  and thus always required (when the site is connected to WordPress.com).

allie500 replied:
I've added this info in [02ef9c5](https://github.com/Automattic/jetpack/pull/43413/commits/02ef9c54feb52bc78ca589fb35dc104229686cdf).

jeherve commented on projects/packages/paypal-payments/src/block/edit.js:
```diff
@@ -0,0 +1,588 @@
+import { __, _n, sprintf } from '@wordpress/i18n';
+import { getWidgetIdFromBlock } from '@wordpress/widgets';
+import clsx from 'clsx';
+import emailValidator from 'email-validator';
+import { get, isEmpty, pick, trimEnd } from 'lodash';
```
Since you're using those external tools, you'd need to bring them as dependencies in the package's `package.json` file.

allie500 replied:
Added `email-validator` dependency in [515815e](https://github.com/Automattic/jetpack/pull/43413/commits/515815e86ee65b6388567ec3e481062a85966c84).

jeherve commented on projects/packages/paypal-payments/src/block/edit.js:
```diff
@@ -0,0 +1,588 @@
+import clsx from 'clsx';
+import emailValidator from 'email-validator';
+import { get, isEmpty, pick, trimEnd } from 'lodash';
+import { getCurrencyDefaults } from '../../shared/currencies';
+import HelpMessage from '../../shared/help-message';
```
You'll need to bring those as well.

allie500 replied:
I don't know if I did this how you would do it so please let me know. I copied the `getCurrencyDefaults` function to the `utils.js` file in the Simple Payments block dir. I then copied the `help-message` JS and SCSS files to the package within the `src/block` directory. I made necessary updates to the `edit.js` file and added the `@automattic/format-currency` dependency. All this was done in [9e281ea](https://github.com/Automattic/jetpack/pull/43413/commits/9e281ea750dccb6b0f1bb46ca92d05d458428e8e).

allie500 replied:
The updates to the `edit.js` file can be found in this commit: [d874ddb](https://github.com/Automattic/jetpack/pull/43413/commits/d874ddbb9a15ddee0b953da86558b750b505e114).

jeherve replied:
I believe that means you'll also need to bring in gridicons as a dependency.

allie500 replied:
🤦  Yep, forgot about that. Fixed in [00bc3d9](https://github.com/Automattic/jetpack/pull/43413/commits/00bc3d936b58e8c79d4c79c8367988b6c6f86ee4).

jeherve commented on projects/packages/paypal-payments/src/block/editor.scss:
```diff
@@ -0,0 +1,146 @@
+@import '@automattic/jetpack-base-styles/gutenberg-base-styles';
```
Since you're using this package, you'll need to add it as a dev dependency.

allie500 replied:
I added it as a dev dependency in [515815e](https://github.com/Automattic/jetpack/pull/43413/commits/515815e86ee65b6388567ec3e481062a85966c84).

jeherve commented on projects/packages/paypal-payments/src/block/featured-media.js:
```diff
@@ -0,0 +1,69 @@
+import { Fragment } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { get } from 'lodash';
+import EditButton from '../../shared/edit-button';
```
You'd need to bring this in as well.

allie500 replied:
I've brought the `edit-button` over in [7341e1a](https://github.com/Automattic/jetpack/pull/43413/commits/7341e1a4beed457c2d1d01bfcaba0ead695f87a4).

jeherve commented on projects/packages/paypal-payments/src/block/product-placeholder.scss:
```diff
@@ -0,0 +1,107 @@
+@import '@automattic/jetpack-base-styles/gutenberg-base-styles';
+@import '../../shared/styles/jetpack-variables';
```
I believe you should be able to replace this to remove the dependency.

allie500 replied:
Thanks for catching this. For this one, I simply replaced the variable call with the value in the `product-placeholder.scss` file and removed the dependency. Let me know if that's what you had in mind. See [15e21df](https://github.com/Automattic/jetpack/pull/43413/commits/15e21df90d306b760c163277e0e8fa14779b7e43).

jeherve commented on projects/packages/paypal-payments/src/block/simple-payments.php:
```diff
@@ -0,0 +1,102 @@
+	}
+
+	// Keep content as-is if rendered in other contexts than frontend (i.e. feed, emails, API, etc.).
+	if ( ! jetpack_is_frontend() ) {
```
This will cause issues once we get rid of the Jetpack. I would recommend adding a `function_exists` check right now to save us some surprises later.

allie500 replied:
Good catch! Added in [b1391f3](https://github.com/Automattic/jetpack/pull/43413/commits/b1391f3187e7307420c0a1ecbda00c3b2d01bb25).

jeherve commented on projects/packages/paypal-payments/src/block/simple-payments.php:
```diff
@@ -0,0 +1,102 @@
+		return $content;
+	}
+
+	$simple_payments = Jetpack_Simple_Payments::get_instance();
```
You'll need to use `Automattic\Jetpack\Paypal_Payments\Simple_Payments` here instead of the class from the Jetpack plugin.

allie500 replied:
Good catch here as well. Fixed in [91f2474](https://github.com/Automattic/jetpack/pull/43413/commits/91f2474b92c094ffca5420d1be9dc27a88678398).

jeherve commented on projects/packages/paypal-payments/src/block/test/validate.js:
```diff
@@ -0,0 +1,29 @@
+import runBlockFixtureTests from '../../../shared/test/block-fixtures';
```
You'd need to bring this as well.

allie500 replied:
I've copied it over in [895859a](https://github.com/Automattic/jetpack/pull/43413/commits/895859a6b6e298b3fa0189522f9b3c310ce1068d).

jeherve commented on projects/packages/paypal-payments/src/block/utils.js:
```diff
@@ -0,0 +1,62 @@
+import { trimEnd } from 'lodash';
+import { getCurrencyDefaults } from '../../shared/currencies';
```
You would need to bring this in too.

allie500 replied:
I forgot to comment on this one yesterday, it was addressed in [9e281ea](https://github.com/Automattic/jetpack/pull/43413/commits/9e281ea750dccb6b0f1bb46ca92d05d458428e8e).

jeherve commented on projects/packages/paypal-payments/src/block/utils.js:
```diff
@@ -0,0 +1,62 @@
+		return formatPriceFallback( price, currency, withSymbol );
+	}
+
+	const siteLocale = window?.Jetpack_Editor_Initial_State?.siteLocale ?? 'en-US';
```
It may be worth adding a to-do item here. Eventually you'll need the package to pass its own data from PHP to JS, instead of relying on the existing Jetpack bundle.

allie500 replied:
This is a good idea. Thanks for suggesting it. I've added the comment in [b5dc87f](https://github.com/Automattic/jetpack/pull/43413/commits/b5dc87f16a2dc962c76289137c49a8c6c1696ea2).

jeherve commented on projects/packages/paypal-payments/src/legacy/class-simple-payments.php:
```diff
@@ -0,0 +1,831 @@
+<?php // phpcs:ignore WordPress.Files.FileName.InvalidClassFileName
```
I think you should be able to remove this now that the name matches the class?

allie500 replied:
Thanks for catching this. I've remove it in [a784c47](https://github.com/Automattic/jetpack/pull/43413/commits/a784c47e2e8765fedab3d193f202ccbd160cc6dc).

jeherve commented on projects/packages/paypal-payments/package.json:
```diff
@@ -21,5 +21,19 @@
 		"build-production-js": "echo 'Not implemented.'",
 		"clean": "true"
 	},
-	"devDependencies": {}
```
You'll need the suite of dev dependencies to build the block (babel, ...). You can check #28630 for an example of when it was done for the Forms package.

allie500 replied:
Thanks for sharing the PR. I reviewed it and the [current dev dependencies for the forms package](https://github.com/Automattic/jetpack/blob/update/move-simple-payments-to-paypal-payments-pkg/projects/packages/forms/package.json#L71). It looks like some that were added in the original PR were later removed, so I'm a bit unsure which ones I should add.

Looking at the current package file for the forms package, I see the following:

Dev dependencies
`@wordpress/babel-plugin-import-jsx-pragma`
`@babel/core`
`@babel/runtime`

I also see this dependency which we may need for this package as well: `@automattic/babel-plugin-replace-textdomain`

Are there any others I'm missing that you would suggest?

jeherve reviewed (commented):
I left a few comments below, hopefully it helps a bit.

*********

You'll need to update `projects/packages/paypal-payments/.gitattributes` so the block's raw files aren't shipped in the stable version of the package. The build directory, however, should be included. Check `projects/packages/forms/.gitattributes` for some examples.

************

You'll need a `babel.config.js` file as well.

allie500 committed:
Replace Jetpack version constant

allie500 committed:
Update .gitattributes
Add Babel config

allie500 commented:
> I left a few comments below, hopefully it helps a bit.
> 
> You'll need to update `projects/packages/paypal-payments/.gitattributes` so the block's raw files aren't shipped in the stable version of the package. The build directory, however, should be included. Check `projects/packages/forms/.gitattributes` for some examples.
> 
> You'll need a `babel.config.js` file as well.

Thanks for all the helpful tips! I've made an initial update to the `.gitattributes` file and added the Babel config file in [22a4d7d](https://github.com/Automattic/jetpack/pull/43413/commits/22a4d7dfbb804899d4c3b131633310e612632432). I will need to circle back to the `.gitattributes` once I get the build stuff nailed down.

allie500 committed:
Add jetpack-blocks dep

allie500 committed:
Add build script and deps

allie500 committed:
Copy getCurrencyDefaults to utils
Copy help-message files to package

allie500 committed:
Add file doc block

allie500 committed:
Update file doc block

allie500 committed:
Update imports

allie500 committed:
Update pnpm-lock.yaml

allie500 committed:
Add edit button
Update import

allie500 committed:
Replace Jetpack variable with the value

allie500 committed:
Add function_exists check

allie500 committed:
Update object instantiation

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Add gridicons dep

allie500 committed:
Update pnpm-lock.yaml

allie500 committed:
Bring in block-fixtures.js and dep
Update validate.js

allie500 committed:
Update pnpm-lock.yaml

allie500 committed:
Add TODO for getting site local

allie500 committed:
Remove PHPCS ignore

allie500 committed:
Move register-jetpack-block into block dir

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Add needed deps to build block
Update webpack config for build

allie500 committed:
Update V2 deprecated edit.js
Use local internal deps

allie500 committed:
Add cache and dist dirs to gitignore

allie500 committed:
Update pnpm-lock.yaml

allie500 committed:
Update Jetpack plugin composer.lock

anomiex commented on projects/packages/paypal-payments/package.json:
```diff
@@ -15,11 +15,48 @@
 	"scripts": {
-		"build": "echo 'Not implemented.'",
+		"build": "pnpm run clean && webpack --config ./webpack.config.blocks.js",
 		"build-js": "echo 'Not implemented.'",
 		"build-production": "echo 'Not implemented.'",
 		"build-production-js": "echo 'Not implemented.'",
```
You can drop `build-js` and `build-production-js` if you're not going to use them. OTOH, you are calling `build-production` from composer.json, so that one should be implemented. It might be as simple as
```suggestion
		"build-production": "NODE_ENV=production BABEL_ENV=production pnpm build",
```

allie500 replied:
Thanks for taking a look! 

> You can drop build-js and build-production-js if you're not going to use them.

I'll go ahead and remove them. If we do end up needing them, it will be easy to re-add them.

> OTOH, you are calling build-production from composer.json, so that one should be implemented. It might be as simple as ...

Yes, thanks for flagging it. I'll get that updated. Based on what I saw in the forms package, your suggested command looks like it should be correct.

allie500 replied:
Addressed in [0735334](https://github.com/Automattic/jetpack/pull/43413/commits/07353346486079ae3752e3b1af07481526b4629f).

anomiex commented on projects/packages/paypal-payments/package.json:
```diff
@@ -15,11 +15,48 @@
+		"autoprefixer": "10.4.20",
+		"postcss": "8.4.47",
+		"postcss-loader": "8.1.1",
+		"sass": "1.64.1",
```
To match #43398,
```suggestion
		"sass": "1.87.0",
```

allie500 replied:
Oh good catch! I missed that `sass` was in the Jetpack plugin package file. It is a dev dependency in the forms package and is using the `1.64.1` version. 

https://github.com/Automattic/jetpack/blob/trunk/projects/packages/forms/package.json#L95

I'll update the version in this package to match Jetpack plugin's version.

allie500 replied:
Updated in [0735334](https://github.com/Automattic/jetpack/pull/43413/commits/07353346486079ae3752e3b1af07481526b4629f).

anomiex replied:
Forms confusingly has two entries for sass 😀
https://github.com/Automattic/jetpack/blob/3c9dbc5bc0de562d5321088ef53bfd1667881d63/projects/packages/forms/package.json#L66
https://github.com/Automattic/jetpack/blob/3c9dbc5bc0de562d5321088ef53bfd1667881d63/projects/packages/forms/package.json#L95

allie500 replied:
ha! That is confusing. Should I put this package's `sass` dependency in the regular deps or in dev deps?

anomiex replied:
For a PHP package it doesn't really matter. Since all the JS gets bundled for the released package anyway, technically they're all dev deps.

allie500 committed:
Update build scripts
Update sass dev-dep

allie500 committed:
Update pnpm-lock.yaml

allie500 committed:
Update phan config and baseline
Add function exists checks

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Update pnpm-lock.yaml

allie500 committed:
Fix Phan issues in Jetpack plugin

jeherve committed:
Update list of files shipped in stable version of the package

jeherve committed:
Remove method that is no longer used

jeherve committed:
Update checks to check if the feature is available

jeherve commented on projects/packages/paypal-payments/src/widget/class-simple-payments-widget.php:
```diff
@@ -0,0 +1,631 @@
+		 */
+		public function form( $instance ) {
+			if ( ! Simple_Payments::is_enabled_jetpack_simple_payments() ) {
+				require __DIR__ . '/simple-payments/admin-warning.php';
+				return;
+			}
+
+			$instance = wp_parse_args( $instance, $this->defaults() );
+
+			$product_posts = get_posts( // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+				array(
+					'numberposts' => 100,
+					'orderby'     => 'date',
+					'post_type'   => Simple_Payments::$post_type_product,
+					'post_status' => 'publish',
+				)
+			);
+
+			require __DIR__ . '/simple-payments/form.php';
```
You'll need to bring in the contents from `projects/plugins/jetpack/modules/widgets/simple-payments/` too, to avoid errors.

allie500 replied:
I've moved the files in [59301fc](https://github.com/Automattic/jetpack/pull/43413/commits/59301fc03935acd77a8b0e1521113efd63a53a00).

jeherve committed:
Move block registration to the package

This will allow us to register the block from the plugin that requires the package (only Jetpack right now, but could be any plugin in the future).

jeherve committed:
Remove block files from the Jetpack plugin

The block is built from the package now.

jeherve reviewed (commented):
I've pushed a few commits to keep moving. At this point, I would recommend testing this in your dev environment or on a test site to see where we still have issues. As long as you have a paid plan like Jetpack Complete on that site (you can purchase one with credits), you should be able to test things.

I've moved the block registration to the package, so you can test the block in the editor and see if everything works properly.

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Move simple payments widget files
Add missing JS Doc comments to customizer
Fix textdomains in php files

allie500 committed:
Update Jetpack plugin phan baseline

allie500 committed:
Update pnpm lockfile

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Update to use sass-embedder

allie500 committed:
Update pnpm lock file

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Fix phan issue in Jetpack plugin
Update phan baseline

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Update pnpm lock file

allie500 committed:
Add watch script

allie500 committed:
Update Jetpack plugin composer lock file

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Add WordPress Blocks dependency

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Update pnpm lock file

allie500 committed:
Remove shared vars merged in from trunk

allie500 committed:
Fix register_block method

allie500 committed:
Fix for WPCOM tests

allie500 committed:
Fix for WPCOM tests

allie500 committed:
Revert "Fix for WPCOM tests"

This reverts commit b6dc30ecf57e63c1e97221df2fb5228624a8e9d4.

allie500 committed:
Revert "Fix for WPCOM tests"

This reverts commit 4729385a49db2b8a9532c691b7a24929e527a733.

allie500 committed:
Update Jetpack_Simple_Payments to delegate all functionality to PayPal Payments Simple_Payments class.

allie500 committed:
Update Jetpack Phan baseline

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Add JS testing deps
Update PNPM lock file
Add JS test script command

allie500 committed:
Add Jest config, setup, etc
Add additiional testing deps
Update PNPM lock file

allie500 committed:
Move JS tests
Update Jest config
Update pnpm js test command

allie500 committed:
Update Jetpack plugin composer.lock

allie500 committed:
Fix composer test-js script

allie500 committed:
Update Jetpack plugin composer.lock

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

millerf commented on projects/plugins/jetpack/modules/simple-payments/simple-payments.php:
```diff
@@ -80,85 +77,24 @@ public static function getInstance() { // phpcs:ignore WordPress.NamingConventio
 		if ( ! self::$instance ) {
 			self::$instance = new self();
-			self::$instance->register_init_hooks();
+			self::$instance->init_hook_action();
```
@allie500 This might be the issue.  `register_init_hooks` used to also call `register_meta_fields_in_rest_api`. Could it be that this is not done anymore?

allie500 replied:
Thanks for flagging this. I don't think this is an issue. The [last line of the class](https://github.com/Automattic/jetpack/blob/update/move-simple-payments-to-paypal-payments-pkg/projects/plugins/jetpack/modules/simple-payments/simple-payments.php#L269) calls the new class's `get_instance` method.

I've also checked by adding some log lines to the two callback methods in `register_init_hooks` and they are logging.
https://github.com/Automattic/jetpack/blob/update/move-simple-payments-to-paypal-payments-pkg/projects/packages/paypal-payments/src/legacy/class-simple-payments.php#L136-L137

```
[30-May-2025 17:50:55 UTC] AJM - Automattic\Jetpack\Paypal_Payments\Simple_Payments::init_hook_action
[30-May-2025 17:50:55 UTC] AJM - Automattic\Jetpack\Paypal_Payments\Simple_Payments::register_meta_fields_in_rest_api
```

allie500 committed:
Remove plan_check as it is not used and breaks rendering

allie500 committed:
Fix PayPal checkout script registration

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Update PNPM lock file

allie500 committed:
Fix frontend CSS not found

allie500 committed:
Remove Jetpack_Options dep

allie500 committed:
Fix webpack config

allie500 committed:
Replace Jetpack_Currencies

allie500 committed:
Uncomment code

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Remove paid plan requirement and checks
Remove Current_Plan & Jetpack_Components deps

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Re-add Jetpack_Options dependency

allie500 committed:
Update Jetpack plugin composer.lock

allie500 committed:
Update get_blog_id()

allie500 committed:
Fix Phan error

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Re-add plan check

allie500 committed:
Add plan check to register_block

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Update PNPM lock file

allie500 committed:
Fix register_script path

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

jeherve commented on projects/packages/paypal-payments/src/block/class-block.php:
```diff
@@ -0,0 +1,158 @@
+		}
+
+		// Keep content as-is if rendered in other contexts than frontend (i.e. feed, emails, API, etc.).
+		if ( function_exists( 'jetpack_is_frontend' ) && ! jetpack_is_frontend() ) {
```
Now that I've merged #43873, you'll have to update this too to use the new class.

allie500 replied:
Thanks for the heads up here. I've updated this in [673d2b9](https://github.com/Automattic/jetpack/pull/43413/commits/673d2b9d5fbdeb402f70ace3a58e233c9c45dc0a).

jeherve commented on projects/packages/paypal-payments/src/widget/class-simple-payments-widget.php:
```diff
@@ -0,0 +1,631 @@
+		 */
+		public function __construct() {
+			parent::__construct(
+				'simple_payments_widget',
```
It looks like you changed the name of the widget and removed the `jetpack_` prefix. This is problematic because some of our code checks for the widget name with the prefix, for example for the actions happening in the `customizer.js` file:
https://github.com/Automattic/jetpack/pull/43413/files#diff-6b5f58c71dbc388e4dd62a6615233829ea9572f7d7186cb4f850ff5ad1ade0feR9

allie500 replied:
This was the source of our issues when testing adding the widget and a new product in legacy mode. I have addressed it in [d5430e2](https://github.com/Automattic/jetpack/pull/43413/commits/d5430e2b23f388dd4c528ee0ee0d8657248c6382) and [5a0b249](https://github.com/Automattic/jetpack/pull/43413/commits/5a0b249d2830c37809a8f884a871ae70ddab3cd6).

I tested on a fresh JN site and it works as expected when following the [steps you outlined here](https://github.com/Automattic/jetpack/pull/43413#pullrequestreview-2916331469).

jeherve reviewed (commented):
We're getting there I think! I found some issues with the legacy widget though. Here is how I would recommend testing it.

1. On a self-hosted site, connect the site to WordPress.com
2. Purchase a paid plan
3. Go to Appearance > Themes, install and activate a classic theme like Twenty Ten
4. Go to Plugins > Add new, install and activate the Classic widgets plugin
5. Go to Appearance > Customize > Widgets
6. Add a Pay with PayPal widget to a widget area
7. Try creating a product from there.

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Use Jetpack Status package
Replace jetpack_is_frontend()

allie500 committed:
Update Jetpack plugin Composer lock file

allie500 committed:
Fix widget name

allie500 committed:
Fix Simple Payments widget nonce logic

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Update lock file

allie500 committed:
Update Jetpack plugin composer lock file

allie500 committed:
Fix ESlint config file

allie500 committed:
Fixing lock file issues

allie500 committed:
Update PNPM lock file

allie500 commented:
Need some help fixing this broken lock file again after merging in trunk. 

It is complaining about `copy-webpack-plugin@11.0.0(webpack@5.94.0)` but I can't get it to update to `copy-webpack-plugin@13.0.0` like that found on trunk.

allie500 committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

allie500 committed:
Update lock file

anomiex committed:
Sync JS dependency versions with the rest of the monorepo

ebinnion committed:
Merge remote-tracking branch 'origin/trunk' into update/move-simple-payments-to-paypal-payments-pkg

ebinnion committed:
PayPal Payments: Fix jetpack-script-data error test

ebinnion commented:
Functionally, this seems to test well for me. Minus the CORS issues on Simple sites.

I've fixed merge conflicts and have fixed a broken test. I think that it's ready for a review.

ebinnion committed:
Merge branch 'trunk' into update/move-simple-payments-to-paypal-payments-pkg

# Conflicts:
#	pnpm-lock.yaml

ebinnion committed:
Fix failing JS tests

ebinnion committed:
Merge trunk

jeherve commented on projects/packages/paypal-payments/src/widget/class-simple-payments-widget.php:
```diff
@@ -0,0 +1,631 @@
+		 * Enqueue styles.
+		 */
+		public function enqueue_style() {
+			wp_enqueue_style( 'simple-payments-widget-style', plugins_url( 'simple-payments/style.css', __FILE__ ), array(), '20180518' );
+		}
+
+		/**
+		 * Enqueue admin styles.
+		 */
+		public function admin_enqueue_styles() {
+			wp_enqueue_style(
+				'simple-payments-widget-customizer',
+				plugins_url( 'simple-payments/customizer.css', __FILE__ ),
```
The widget scripts and styles do not seem to load anymore. I think the location of the files should be updated for this to work, maybe.

ebinnion replied:
Thank you sir. I know that I tested the widget, but I missed this. I'll look into it.

ebinnion replied:
Interestingly, it seems to be working in my local Jetpack development environment but not in a build that is pushed to a WoW site.

ebinnion replied:
That CSS class seems to load fine. The issue seems to be that the legacy simple-payments.css doesn't get enqueued.

ebinnion replied:
I tracked the issue down. I hadn't selected a product correctly so the widget wasn't loading on the frontend correctly. 🫠

ebinnion replied:
I haven't been able to reproduce the issue. Are you sure the styles and scripts aren't loading? The biggest questions that I have are why is there no image or "Buy this!" header in the before, but there is in the after? But, I don't think that this would be an issue with the scripts and styles loading?

jeherve reviewed (commented):
I don't notice anything wrong with the block, but the legacy widget doesn't seem to be styled properly anymore:

**Before**

<img width="451" alt="Image" src="https://github.com/user-attachments/assets/897781ed-4862-43cf-88aa-21bd24d52e7a" />

**After**

<img width="382" alt="Image" src="https://github.com/user-attachments/assets/5e75cfc3-5bb1-4342-85fe-6c1028d7de88" />

ebinnion committed:
Fix issue when NumberFormatter class does not exist

ebinnion committed:
Fix phan issue

jeherve reviewed (dismissed):
As we discussed in Slack, this is in a good state for a merge. There are some display issues in the legacy widget still, but that can be addressed in a follow-up PR.

Of note, when merging this PR, it will get deployed to WordPress.com simple right after the merge, so we must be ready for feedback right then.
Deployments to WoA happen on Mondays, so that's also something to keep in mind ; you wouldn't want to deploy this to Atomic without the follow-up I think.

ebinnion committed:
Merge from trunk
